server.js

```diff
@@ -0,0 +1,341 @@
+              else {
+                let bodyData = querystring.parse(body);
+                let newPageTemplate = `
+                  <!DOCTYPE html>
+                  <html lang="en">
+    
+                  <head>
+                    <meta charset="UTF-8">
+                    <title>The Elements - ${bodyData.elementName}</title>
+                    <link rel="stylesheet" href="/css/styles.css">
+                  </head>
+    
+                  <body>
+                    <h1>${bodyData.elementName}</h1>
+                    <h2>${bodyData.elementSymbol}</h2>
+                    <h3>Atomic number ${bodyData.elementAtomicNumber}</h3>
+                    <p>${bodyData.elementDescription}</p>
+                    <p><a href="/">back</a></p>
+                  </body>
+    
+                  </html>
+                `;
+
+                fs.writeFile(`./public/${bodyData.elementName.toLowerCase()}.html`, newPageTemplate, (err, data) => {
+                  if (err) {
+                    res.writeHead(500);
+                    return res.end('{ "success": false }');
+                  }
+
+                  res.writeHead(200, { 'Content-Type': 'application/json' });
+                  return res.end('{ "success": true }');
+                });
+              }
+            });
+          } else if (err.code === "ENOENT") {
+            res.writeHead(500, { "Content-Type": "application/json" });
+            res.end(`{ "error" : "resource ${req.url} does not exist" }`);
+          } else {
+            res.writeHead(err.status, { "Content-Type": "application/json" });
+            res.end(`{ "Error" : ${err.code} }`);
+          }
+        });
+      } else {
+        res.writeHead(401, { "WWW-Authenticate": 'Basic realm="SecureArea"' });
+        return res.end(`<html><body>Invalid Authentication Credentials</body></html>`);
+      }
+
+    } else {
+      res.writeHeader(401, { 'WWW-Authenticate': 'Basic realm="SecureArea"' });
+      return res.end(`<html><body>Not Authorized</body></html>`);
+    }
+
+    /*
+    * DELETE METHOD
+    */
+
+  } else if (req.method === "DELETE") {
+    if (req.headers.authorization) {
+      const encodedString = req.headers.authorization.slice(6);
+      const base64Buffer = new Buffer(encodedString, 'base64');
+      const decodedString = base64Buffer.toString();
+      let user = decodedString.slice(0, decodedString.indexOf(':'));
+      let pass = decodedString.slice(decodedString.indexOf(':') + 1);
+
+      if (authorized.hasOwnProperty(user) && authorized[user] === pass) {
+        fs.stat(`./public/${req.url}`, (err, stats) => {
+          if (err == null) {
+            fs.unlink(`./public/${req.url}`, (err) => {
+              if (err) {
+                res.writeHead(500, { "Content-Type": "application/json" });
+                res.end(`{ "Server error" : "Could not delete file" }`);
+              } else {
+                fs.readdir('./public', (err, files) => {
```

Comment:
This section can be turned into a function to basically handle reading in a dir and returning the list of files, this function could even take in a blacklist and filter out and return only the files that are element related.